### PR TITLE
Fix migrate_db_chrysalis_to_stardust() with ledger nano signer type

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Fix migrate_db_chrysalis_to_stardust() with ledger nano signer type

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Added test

### CODE REVIEW

File: sdk/src/wallet/migration/chrysalis.rs [hunk 1]
```diff
@@ -167,8 +167,8 @@ pub(crate) fn migrate_from_chrysalis_data(
                                 r#"{{"Stronghold": {{"password": null, "timeout": null, "snapshotPath": "{}/wallet.stronghold"}} }}"#,
                                 storage_path.to_string_lossy()
                             ),
-                            Some("LedgerNano") => r#"{{"LedgerNano": false }}"#.into(),
-                            Some("LedgerNanoSimulator") => r#"{{"LedgerNano": true }}"#.into(),
+                            Some("LedgerNano") => r#"{"LedgerNano": false }"#.into(),
```
Comment (DaughterOfMars) [inline sdk/src/wallet/migration/chrysalis.rs]: any reason not to use `json!` macro here?

File: sdk/tests/wallet/chrysalis_migration.rs [hunk 2]
```diff
@@ -384,3 +384,61 @@ fn copy_folder(src: impl AsRef<Path>, dest: impl AsRef<Path>) -> io::Result<()>
...
+    assert_eq!(alice_acc_details.internal_addresses().len(), 1);
+    assert_eq!(
+        alice_acc_details.internal_addresses()[0]
+            .address()
+            .try_to_bech32("rms")?,
+        "rms1qzev23h8qtdfjzzx4jqrdfaw2nnnwu2m4hhu2tkdmp2wrt6y8qwq22963tv"
+    );
+
+    let chrysalis_data = wallet.get_chrysalis_data().await?.unwrap();
+    let accounts_indexation = chrysalis_data.get("iota-wallet-account-indexation").unwrap();
```
Comment (DaughterOfMars) [inline sdk/tests/wallet/chrysalis_migration.rs]: no constant?

File: sdk/src/wallet/migration/chrysalis.rs (hunk 1 above)
Comment (Thoralf-M) [reply sdk/src/wallet/migration/chrysalis.rs]: No, changed

File: sdk/tests/wallet/chrysalis_migration.rs (hunk 2 above)
Comment (Thoralf-M) [reply sdk/tests/wallet/chrysalis_migration.rs]: Added

File: sdk/tests/wallet/chrysalis_migration.rs [hunk 3]
```diff
@@ -384,3 +386,61 @@ fn copy_folder(src: impl AsRef<Path>, dest: impl AsRef<Path>) -> io::Result<()>
...
     Ok(())
 }
+
+#[cfg(feature = "ledger_nano")]
+#[tokio::test]
+async fn migrate_chrysalis_db_ledger() -> Result<()> {
+    let logger_output_config = fern_logger::LoggerOutputConfigBuilder::new()
+        .name("chrysalis.log")
+        .target_exclusions(&["h2", "hyper", "rustls"])
+        .level_filter(log::LevelFilter::Debug);
```
Comment (thibault-martinez) [inline sdk/tests/wallet/chrysalis_migration.rs]: Do we want to keep it ? May be a bit spammish

File: sdk/tests/wallet/chrysalis_migration.rs [hunk 4]
```diff
@@ -384,3 +386,61 @@ fn copy_folder(src: impl AsRef<Path>, dest: impl AsRef<Path>) -> io::Result<()>
...
+
+    fern_logger::logger_init(config).unwrap();
+
+    let storage_path = "migrate_chrysalis_db_ledger/db";
+    setup(storage_path)?;
+    // Copy db so the original doesn't get modified
+    copy_folder("./tests/wallet/fixtures/chrysalis-db-ledger/db", storage_path).unwrap();
+
+    migrate_db_chrysalis_to_stardust("migrate_chrysalis_db_ledger", None, None).await?;
+    println!("hier");
```
Comment (thibault-martinez) [inline sdk/tests/wallet/chrysalis_migration.rs]: :trollface:

File: sdk/tests/wallet/chrysalis_migration.rs (hunk 3 above)
Comment (Thoralf-M) [reply sdk/tests/wallet/chrysalis_migration.rs]: No, removed

File: sdk/tests/wallet/chrysalis_migration.rs (hunk 4 above)
Comment (Thoralf-M) [reply sdk/tests/wallet/chrysalis_migration.rs]: :trollface:

File: sdk/src/wallet/migration/chrysalis.rs [hunk 5]
```diff
@@ -146,9 +144,9 @@ pub(crate) fn migrate_from_chrysalis_data(
     storage_path: &Path,
     // in stronghold the keys are hashed first
     stronghold: bool,
-) -> Result<(Vec<AccountDetailsDto>, Option<String>)> {
+) -> Result<(Vec<AccountDetailsDto>, Option<Value>)> {
     let mut new_accounts: Vec<AccountDetailsDto> = Vec::new();
-    let mut secret_manager_dto: Option<String> = None;
+    let mut secret_manager_dto: Option<Value> = None;
```
Comment (thibault-martinez) [inline sdk/src/wallet/migration/chrysalis.rs:149]: We could return to_string from this function directly ? Bit weird to return a Value. Or do we need to access its fields ?

Comment (Thoralf-M) [reply sdk/src/wallet/migration/chrysalis.rs:149]: In L 127 we don't need the extra conversion if we return Value

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**